### PR TITLE
Remove `FileInfo`, `FileKey`

### LINKED ISSUE #1206: Store files with ID in path
Now we store files and test suites source with organization and project names in path.
Need to use IDs in path to allow renaming projects, organizations and test suites source without moving files

### PR DESCRIPTION
**Pull request by nulls**
### What's done:
- [x] #1672 -- now it's an abstract class which can be extended by DTO which wants to store ID from database
- [x] #1674
- [x] #1680
- [x] #1679
- [x] #1697
- [x] #1699
- [x] #1683
- [x] removed `FileInfo`, `FileKey`

It closes #1206

### CODE REVIEW

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/domain/SandboxFileInfo.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import kotlinx.serialization.Serializable
+
+/**
+ * Class that contains metadata of Sandbox files
+ *
+ * @property sizeBytes size in bytes
+ * @property name name of a file
+ */
+@Serializable
+data class SandboxFileInfo(
```
Comment (Cheshiriks) [inline save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/domain/SandboxFileInfo.kt:12]: Will we not make a File table for the sandbox?

Comment (nulls) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/domain/SandboxFileInfo.kt:12]: should do, will create an issue for it

Comment (nulls) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/domain/SandboxFileInfo.kt:12]: https://github.com/saveourtool/save-cloud/issues/1718

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/storage/FileStorage.kt [hunk 2]
```diff
@@ -1,71 +1,74 @@
...
+    override fun createNewEntityFromDto(dto: FileDto): File = dto.toEntity {
+        projectService.findByNameAndOrganizationNameAndCreatedStatus(dto.projectCoordinates.projectName, dto.projectCoordinates.organizationName)
+            .orNotFound {
+                "Not found project by coordinates: ${dto.projectCoordinates}"
+            }
+    }
 
-    @Suppress(
-        "DestructuringDeclarationWithTooManyEntries"
+    override fun findByDto(dto: FileDto): File? = fileRepository.findByProject_Organization_NameAndProject_NameAndNameAndUploadedTime(
```
Comment (kgevorkyan) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/storage/FileStorage.kt:39]: nameandnameandname 😄

**Review by kgevorkyan — APPROVED** [review]
lgtm

**Pull request merged**